ckanext/canada/helpers.py

```diff
@@ -719,6 +728,70 @@ def flash_success(message, allow_html=True):
               allow_html=allow_html)
 
 
+def get_loader_status_badge(resource):
+    # type: (dict) -> str
+    """
+    Displays a custom badge for the status of Xloader and DataStore
+    for the specified resource.
+    """
+
+    if not XLoaderFormats:
+        return ''
+
+    if not resource.get('url_type') == 'upload' or \
+    not XLoaderFormats.is_it_an_xloader_format(resource.get('format')):
+        # we only want to show badges for uploads of supported xloader formats
+        return ''
+
+    is_datastore_active = resource.get('datastore_active', False)
+
+    try:
+        xloader_job = t.get_action("xloader_status")(None, {"resource_id": resource.get('id')})
+    except (t.ObjectNotFound, t.NotAuthorized):
+        xloader_job = {}
+
+    if xloader_job.get('status') == 'complete':
+        # the xloader task is complete, show datastore active or inactive.
+        # xloader will delete the datastore table at the beggining of the job run.
+        # so this will only be true if the job is fully finished.
+        status = 'active' if is_datastore_active else 'inactive'
+    elif xloader_job.get('status') in ['pending', 'running', 'running_but_viewable', 'error']:
+        # the job is running or pending or errored
+        # show the xloader status
+        status = xloader_job.get('status')
+        if status == 'running_but_viewable':
+            # treat running_but_viewable the same as running
+            status = 'running'
+    else:
+        # we do not know what the status is
+        status = 'unknown'
+
+    messages = {
+        'pending': _('Data awaiting load to DataStore'),
+        'running': _('Loading data into DataStore'),
+        'complete': _('Data loaded into DataStore'),
+        'error': _('Failed to load data into DataStore'),
+        'active': _('Data available in DataStore'),
+        'inactive': _('Resource not active in DataStore'),
+        'unknown': _('DataStore status unknown'),
+    }
+
+    pusher_url = t.h.url_for('xloader.resource_data',
+                             id=resource.get('package_id'),
+                             resource_id=resource.get('id'))
+
+    badge_url = t.h.url_for_static('/static/img/badges/{lang}/datastore-{status}.svg'.format(lang=t.h.lang(), status=status))
+
+    title = t.h.render_datetime(xloader_job.get('last_updated'), with_hours=True) \
+        if xloader_job.get('last_updated') else ''
+
+    return Markup(u'<a href="{pusher_url}" class="loader-badge"><img src="{badge_url}" alt="{alt}" title="{title}"/></a>'.format(
+        pusher_url=pusher_url,
+        badge_url=badge_url,
+        alt=html.escape(messages[status].replace('"', '\"')),
+        title=html.escape(title.replace('"', '\"'))))
```

Comment:
doesn't https://docs.python.org/3/library/html.html#html.escape already handle quotes? Maybe only needs `quote=True` for python 2 compatibility